test/unit/communication/test_reduce.cpp

```diff
@@ -8,55 +8,207 @@
+}
 
-  int value = 1;
-  int result = 0;
+using TypeParam = std::complex<double>;
 
-  sync::reduce(root, world, MPI_SUM, make_message(common::make_data(&value, 1)),
-               make_message(common::make_data(&result, 1)));
+TEST_F(ReduceTest, ValueOnSingleRank) {
+  CommunicatorGrid alone_grid(world, 1, 1, common::Ordering::RowMajor);
 
-  // check that the root rank has the reduced results
-  if (world.rank() == root) {
-    EXPECT_EQ(NUM_MPI_RANKS * value, result);
-  }
+  Communicator alone_world = alone_grid.rowCommunicator();
 
-  // check that input has not been modified
-  EXPECT_EQ(1, value);
+  // just the master rank has to reduce
+  if (alone_world == MPI_COMM_NULL)
+    return;
+
+  constexpr int root = 0;
+  constexpr TypeParam value = dlaf_test::TypeUtilities<TypeParam>::element(13, 26);
+  TypeParam result = 0;
+
+  ASSERT_EQ(alone_world.rank(), root);
+
+  sync::reduce(root, alone_world, MPI_SUM, common::make_data(&value, 1), common::make_data(&result, 1));
+
+  DLAF_EXPECT_NEAR(value, result, TypeUtilities<TypeParam>::error);
 }
 
-TEST_F(ReduceTest, Multiple) {
+TEST_F(ReduceTest, CArrayOnSingleRank) {
+  CommunicatorGrid alone_grid(world, 1, 1, common::Ordering::RowMajor);
+
+  Communicator alone_world = alone_grid.rowCommunicator();
+
+  // just the master rank has to reduce
+  if (alone_world == MPI_COMM_NULL)
+    return;
+
+  constexpr int root = 0;
+  constexpr std::size_t N = 3;
+  constexpr TypeParam input[N] = {dlaf_test::TypeUtilities<TypeParam>::element(0, 1),
+                                  dlaf_test::TypeUtilities<TypeParam>::element(1, 2),
+                                  dlaf_test::TypeUtilities<TypeParam>::element(2, 3)};
+  TypeParam reduced[N];
+
+  ASSERT_EQ(alone_world.rank(), root);
+
+  sync::reduce(root, alone_world, MPI_SUM, common::make_data(input, N), common::make_data(reduced, N));
+
+  for (std::size_t index = 0; index < N; ++index)
+    DLAF_EXPECT_NEAR(input[index], reduced[index], TypeUtilities<TypeParam>::error);
+}
+
+TEST_F(ReduceTest, Value) {
+  const int root = 0;
+  constexpr TypeParam value = dlaf_test::TypeUtilities<TypeParam>::element(13, 26);
+  TypeParam result = 0;
+
+  sync::reduce(root, world, MPI_SUM, common::make_data(&value, 1), common::make_data(&result, 1));
+
+  if (world.rank() == root)
+    DLAF_EXPECT_NEAR(value * static_cast<TypeParam>(NUM_MPI_RANKS), result,
+                     NUM_MPI_RANKS * TypeUtilities<TypeParam>::error);
+}
+
+TEST_F(ReduceTest, CArray) {
   static_assert(NUM_MPI_RANKS >= 2, "This test requires at least two ranks");
   int root = 1;
 
-  const std::size_t N = 3;
-  int input[N] = {1, 2, 3};
-  int reduced[N];
+  constexpr std::size_t N = 3;
+  constexpr TypeParam input[N] = {dlaf_test::TypeUtilities<TypeParam>::element(0, 1),
+                                  dlaf_test::TypeUtilities<TypeParam>::element(1, 2),
+                                  dlaf_test::TypeUtilities<TypeParam>::element(2, 3)};
```

Comment:
There are few types and function used throughout the test, I'd encourage a heavier use of `using` declarations:

```
using dlaf_test::TypeUtilities<TypeParam>::element;
```

Reply:
I'm not sure that I can do what you are proposing, since `element` is a static member for the template class `TypeUtilities`.

In case you confirm that, I can propose you:
- just remove the `dlaf_test` since there is the `using dlaf_test` directive at the beginning of the file;
- for each test delcare `using TypeUtilities = typename TypeUtilities<TypeParam>` in order to create a "shortcut" alias

Feel free to suggest your solution 😉

Reply:
Both seem fine to me, it's not a big issue in the grand scheme of things :) . You can also try this:

`using TypeUtils = dlaf_test::TypeUtilities<TypeParam>` and then do `TypeUtils::element()` 

if you prefer.